tests/pytorch/test_pytorch_autolog.py

```diff
@@ -0,0 +1,168 @@
+import os
+import pytest
+import pytorch_lightning as pl
+import torch
+from iris import IrisClassification
+import mlflow
+import mlflow.pytorch
+from mlflow.pytorch.pytorch_autolog import autolog
+from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+from pytorch_lightning.callbacks import ModelCheckpoint
+
+NUM_EPOCHS = 20
+
+
+@pytest.fixture
+def pytorch_model():
+    autolog()
+    model = IrisClassification()
+    trainer = pl.Trainer(max_epochs=NUM_EPOCHS)
+    trainer.fit(model)
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(client.list_run_infos(experiment_id="0")[0].run_id)
+    return trainer, run
+
+
+def test_pytorch_autolog_logs_default_params(pytorch_model):
+    _, run = pytorch_model
+    data = run.data
+    assert "learning_rate" in data.params
+    assert "epsilon" in data.params
+    assert "optimizer_name" in data.params
+
+
+def test_pytorch_autolog_logs_expected_data(pytorch_model):
+    _, run = pytorch_model
+    data = run.data
+
+    # Checking if metrics are logged
+    assert "loss" in data.metrics
+    assert "val_loss" in data.metrics
+
+    # Testing unwanted parameters are not logged
+    assert "callbacks" not in data.params
+
+    # Testing optimizer parameters are logged
+    assert "optimizer_name" in data.params
+    assert data.params["optimizer_name"] == "Adam"
+
+    # Testing model_summary.txt is saved
+    client = mlflow.tracking.MlflowClient()
+    artifacts = client.list_artifacts(run.info.run_id)
+    artifacts = map(lambda x: x.path, artifacts)
+    assert "model_summary.txt" in artifacts
+
+
+@pytest.fixture
+def pytorch_model_initialized():
+    trainer = pl.Trainer(max_epochs=NUM_EPOCHS)
+    model = IrisClassification()
+    trainer.fit(model)
+    return trainer
+
+
+def test_pytorch_autolog_persists_manually_created_run(pytorch_model_initialized):
+    _ = pytorch_model_initialized
```

Comment:
```suggestion
    _ = pytorch_model_initialized
```

should we remove this line?

Reply:
pylint throws up the unused argument warning without this line. I removed the line and added module level disable for it.